### PR TITLE
hibernate validator: Remember annotated interface methods

### LINKED ISSUE #1888: Quarkus cannot use the hibernate-validator constraints in the inherited class/interface
If you use constraints from the hibernate-validator like [here](https://quarkus.io/guides/validation-guide#rest-end-point-validation) but in a parent interface (or class) the constraints don't work.

### Scenario
**ExampleResourceImpl** is implementing **ExampleResource** (it's a interface).  
**ExampleResource** has all `@GET` and `@Path` (JAX-RS) annotations and has also all hibernate validation constraints like `@Valid` `@NotNull`.  

ExampleResourceImpl.java
```java
@RequestScoped
public class ExampleResourceImpl implements ExampleResource {

    public Response doSomething(SomeObject someObject) {
        return something();
    }
}
```

ExampleResource.java
```java
@Produces(APPLICATION_JSON)
public interface ExampleResource extends Serializable {

    @POST
    @Path("/example")
    Response doSomething(@Valid @NotNull SomeObject someObject);

}
```

### Summary
Quarkus can read the JAX-RS annotations, **but not** the hibernate-validator **constraints** from his parent class/interface.

### PR DESCRIPTION
**Pull request by ebullient**
resolves #1888

This includes #1913

### CODE REVIEW

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java [hunk 1]
```diff
@@ -245,6 +254,20 @@ private static void contributeClassMarkedForCascadingValidation(Set<DotName> cla
...
+    private static void contributeMethodsWithInheritedValidation(Map<DotName, Set<String>> inheritedAnnotationsToBeValidated,
+            IndexView indexView, MethodInfo method) {
+        ClassInfo clazz = method.declaringClass();
+        if (Modifier.isInterface(clazz.flags())) {
+            // Remember annotated interface methods that must be validated
+            Set<String> methodSet = inheritedAnnotationsToBeValidated.get(clazz.name());
+            if (methodSet == null) {
+                methodSet = new HashSet<String>();
+                inheritedAnnotationsToBeValidated.put(clazz.name(), methodSet);
+            }
```
Comment (aguibert) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java]: could collapse this down a bit using `computeIfAbsent()`

**Review by aguibert — APPROVED** [review]

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java (hunk 1 above)
Comment (ebullient) [reply extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java]: it isn't a concurrent set

**Comment by gsmet** [comment]
I'm away for a few days. I'll have a look when I'll get back.

**Comment by geoand** [comment]
I'll also give it a look over the weekend but I'll leave the final approval to @gsmet since he's the sme on this

**Comment by geoand** [comment]
Nice job!

I do see however that the original issue mentions inherited methods from classes as well as interfaces and unless I misunderstood the changes, the class case is not handled?

**Comment by ebullient** [comment]
> Nice job!
> 
> I do see however that the original issue mentions inherited methods from classes as well as interfaces and unless I misunderstood the changes, the class case is not handled?

I will double check. I'll have to add an additional test, I think. I stuck with the testcase provided by @isikemre, which focused on interfaces.

**Comment by geoand** [comment]
Cool, thanks!

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
it looks it introduces issues in other PR (CI builds), I'm going to revert it, and @gsmet will have another look

**Comment by gsmet** [comment]
```
ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.453 s <<< FAILURE! - in io.quarkus.it.hibernate.validator.HibernateValidatorFunctionalityTest
[ERROR] testInheritedImplementsConstraints  Time elapsed: 0.047 s  <<< FAILURE!
java.lang.AssertionError:
1 expectation failed.
Response body doesn't match expectation.
Expected: is "passed\nfailed: echoZipCode.arg0 (size must be between 5 and 5)"
```

**Comment by geoand** [comment]
So there is flaky test

**Comment by geoand** [comment]
Question: Did the test only fail in native mode?

**Comment by gsmet** [comment]
The one I saw was in native.

**Pull request merged**